### DISCUSSION TITLE
Any ideas to implement East / West facing stairs movement?

### DISCUSSION THREAD

**Discussion by jahvi** [Annoraaq/grid-engine#496] [Q&A]
I was wondering if anyone has implemented grid engine using east and west facing stairs, so far it seems like more trouble than it's worth but maybe there's a simple workaround I'm missing?

Right now the best idea I have is to enable 8 way movement and design the stair tiles in a way that they're always 45deg so the character can move diagonally from top to bottom (or viceversa), the main issue is that I only want to enable diagonals on these types of stairs and not anywhere else in the map and I'm not sure if that's possible.

**Comment by Annoraaq** [comment]
Hey, so I have no better idea at the moment. You can enable 8 directional movement and then disable it for almost all tiles which is a bit cumbersome but you could write a script for it that does it either on your tilemaps JSON files or when you initialize the game. You need to use [one-way collisions](https://annoraaq.github.io/grid-engine/p/tile-properties/#one-way) on your tiles. Maybe someone has a better idea?
